prereise/gather/griddata/transmission/geometry.py

```diff
@@ -250,6 +251,58 @@ def calculate_shunt_capacitance_per_km(self):
         return capacitance_per_km
 
 
+@dataclass
+class Line(DataclassWithValidation):
+    """Given a Tower design, line voltage, and length, calculate whole-line impedances
+    and rating.
+
+    :param Tower tower: tower parameters (containing per-kilometer impedances).
+    :param int/float length: line length (kilometers).
+    :param int/float voltage: line voltage (kilovolts).
+    :param int/float freq: the system nominal frequency (Hz).
+    """
+
+    tower: Tower
+    length: float
+    voltage: float
+    freq: float = 60.0
+    series_impedance_per_km: complex = field(init=False)
+    shunt_admittance_per_km: complex = field(init=False)
+    propogation_constant_per_km: complex = field(init=False)
+    surge_impedance: complex = field(init=False)
+    series_impedance: complex = field(init=False)
+    shunt_admittance: complex = field(init=False)
+
+    def __post_init__(self):
+        # Convert integers to floats as necessary
+        for attr in ("freq", "length", "voltage"):
+            if isinstance(getattr(self, attr), int):
+                setattr(self, attr, float(getattr(self, attr)))
+        self.validate_input_types()  # defined in DataclassWithValidation
+        # Calculate second-order electrical parameters which depend on frequency
+        self.omega = 2 * pi * self.freq
```

Comment:
Not sure whether we would like to define `omega` as an attribute of a `Line` object or it is just something intermediate.

Reply:
Good point, it's only really intermediate, and anyone who would want to look at the value should know how to calculate it manually. Removed.